### PR TITLE
Add New drive types

### LINKED ISSUE #53: Add support for new drive types
Differential drive, Killough Drive, and Mecanum, as well as SpeedControllerGroup.

### PR DESCRIPTION
**Pull request by sciencewhiz**
Fixes #53 

- [x] SpeedControllerGroup
- [x] Differential Drive
- [x] Mecanum Drive
- [x] Killough Drive

### CODE REVIEW

**Comment by sciencewhiz** [comment]
Java works. C++ either needs a change to wpilib SpeedControllerGroup to accept shared pointers (like robot drive does) or Robot Builder needs big changes to use references instead of shared pointers.

**Comment by sciencewhiz** [comment]
Changes were not that big on the robot builder side. Thanks @calcmogul

File: src/main/resources/help/Speed Controller.html [hunk 1]
```diff
@@ -22,9 +22,12 @@
       <dd>The type of speed controller to use.</dd>
       <dt>Output Port (PWM)</dt>
       <dd>The PWM port on the roboRIO or MXP that the output is plugged into.</dd>
+      <dt>Inverted</dt>
+      <dd>Whether to invert the direction of the speed controller.</dd>
     </dl>
 
 
+
```
Comment (PeterJohnson) [inline src/main/resources/help/Speed Controller.html]: 3 blank lines?  Other files look like they only have 1.

File: src/main/resources/help/Killough Drive.html [hunk 2]
```diff
@@ -0,0 +1,44 @@
+<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en">
+  <head>
+    <link rel="stylesheet" href="styles.css" type="text/css" media="screen" />
+  </head>
+
+  <body>
+      <h1>Killough Drive</h1>
+
+      <center><img src="../icons/Killough Drive.png" /></center>
```
Comment (PeterJohnson) [inline src/main/resources/help/Killough Drive.html]: Inconsistent indentation

File: src/main/resources/help/Differential Drive.html [hunk 3]
```diff
@@ -0,0 +1,45 @@
+<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en">
+  <head>
+    <link rel="stylesheet" href="styles.css" type="text/css" media="screen" />
+  </head>
+
+  <body>
+      <h1>Differential Drive</h1>
+
+      <center><img src="../icons/Differential Drive.png" /></center>
```
Comment (PeterJohnson) [inline src/main/resources/help/Differential Drive.html]: Inconsistent indentation

File: src/main/resources/help/Speed Controller Group.html [hunk 4]
```diff
@@ -0,0 +1,38 @@
+<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en">
+  <head>
+    <link rel="stylesheet" href="styles.css" type="text/css" media="screen" />
+  </head>
+
+  <body>
+      <h1>Speed Controller Group</h1>
+
+      <center><img src="../icons/Speed Controller Group.png" /></center>
```
Comment (PeterJohnson) [inline src/main/resources/help/Speed Controller Group.html]: Inconsistent indentation

**Pull request merged**